romancal/patch_match/patch_match.py

```diff
@@ -0,0 +1,291 @@
+"""
+This module determines which patch files overlap with the given image.
+
+Currently this assumes that the sky projected borders of all images are straight.
+"""
+
+import os
+import os.path
+
+import asdf
+import gwcs.wcs as wcs
+import numpy as np
+import spherical_geometry.polygon as sgp
+import spherical_geometry.vector as sgv
+from matplotlib import pyplot as plt
+
+RAD_TO_ARCSEC = 180.0 / np.pi * 3600.0
+
+plt.ion()
+
+PATCH_TABLE = None
```

Comment:
The use of a global is something I would also like to avoid, and I'm sure this can be done better. Though note that the idea here is avoiding reloading the table multiple times if multiple images are to be processed. This global could be encapsulated in higher level function that works on an association, or as a class attribute.